services/oauth/v2/destination_info.go

```diff
@@ -0,0 +1,39 @@
+package v2
+
+import "github.com/rudderlabs/rudder-server/utils/misc"
+
+type DestinationInfo struct {
+	WorkspaceID   string
+	DestDefName   string
+	DestDefConfig map[string]interface{}
+	DestinationId string
+	DestConfig    map[string]interface{}
+}
+
+func (d *DestinationInfo) IsOAuthDestination() bool {
+	if authValue, err := misc.NestedMapLookup(d.DestDefConfig, "auth", "type"); err == nil {
+		if authType, ok := authValue.(string); ok {
+			return authType == "OAuth"
+		}
+	}
+	return false
+}
```

Comment:
WE can probably remove one level of nesting and return early.
```suggestion
func (d *DestinationInfo) IsOAuthDestination() bool {
	authValue, err := misc.NestedMapLookup(d.DestDefConfig, "auth", "type")
	if err != nil {
		return false
	}
	if authType, ok := authValue.(string); ok {
		return authType == "OAuth"
	}
	return false
}
```